azurerm/internal/services/compute/linux_virtual_machine_resource.go

```diff
@@ -568,7 +568,11 @@ func resourceLinuxVirtualMachineRead(d *schema.ResourceData, meta interface{}) e
 
 	if profile := props.OsProfile; profile != nil {
 		d.Set("admin_username", profile.AdminUsername)
-		d.Set("allow_extension_operations", profile.AllowExtensionOperations)
+		allowExtensionOperations := true // this is the default value of allowExtensionOperations. If the service returns null for this property, it should mean `true`
```

Comment:
unfortunately since there's security implications to this (e.g. this says that arbitary scripts can be executed on the VM) - defaulting this to `true` is misleading here - if this isn't being returned from the API unfortunately we'll need to raise a bug with the Azure API to get this returned